### ISSUE TITLE
Rethink library language saving

### ISSUE THREAD

**Issue by emehri** [kiwix/kiwix-desktop#902]
GoldenDict and Kiwix 2.3.0 don't support some ZIMs like wikipedia_es_all_nopic_2022-05.zim
But the Kiwix plugin in Firefox can.
How can I view them in those applications?

**Comment by kelson42** [comment]
@emehri We are not in charge of Goldendict here and Kiwix Desktop shoukd be fine. What is exactly the problem, please the exact reproduction steps.

**Comment by kelson42** [comment]
@emehri Sorry I don't understand your comment, I don't understand what is wrong.

**Comment by tim78245** [comment]
Kiwix has this bizarre practice of only showing the default language in 'local files'. I don't think there's an option to change that behavior. Try clicking 'Browse by Language > All' or 'Browse by Language > Spanish' in the sidebar.

**Comment by kelson42** [comment]
@tim78245 So the problem is about the UI/UX of the library. It shows per default, the language used in the OS and you would like to be able to change to another one (and keep the setting)?

**Comment by tim78245** [comment]
> @tim78245 So the problem is about the UI/UX of the library. It shows per default, the language used in the OS and you would like to be able to change to another one (and keep the setting)?

I would say it would be less confusing to simply show all by default, considering the UI doesn't even tell that you're only viewing the system language by default. When I open Kiwix I'm seeing "Local Files" bolded in the sidebar, not "Browse by Language".

**Comment by gmikhail** [comment]
I still believe that showing by default only ZIM files in OS language is a bug (you can call it "unexpected behavior for user" if that's easier for you). All filters should be reset by default. This is the law. Like variables i and j in a loop. 

Quoting this post: https://github.com/kiwix/kiwix-desktop/issues/911#issuecomment-1366473686
> * If changed then this should be kept

> for kiwix-desktop it works already as expexted

No, it doesn't work. Each time you start Kiwix, the language filter is set to the OS language. And the changes are not saved after relaunch.

**Comment by kelson42** [comment]
See #974 for remembering values over restart of Kiwix

**Comment by kelson42** [comment]
I disagree to display all languages per default, default behaviour/configuration should be adapted to user if possible. Guessing the language from the OS is a good approach because using an OS in a language you don't understand simply does not make sense.

For the rest, I agree with most of the things and here are the tickets tackling things:
* https://github.com/kiwix/kiwix-desktop/issues/974 for remembering values over restart of Kiwix
* [#958](https://github.com/kiwix/kiwix-desktop/issues/958) for multiple and visible language selection

**Comment by gmikhail** [comment]
> I disagree to display all languages per default

I don't agree with this. Let me explain my position: 

When a user downloads a ZIM file in their native language (because there is more complete local information) and a ZIM file in English (because there is more complete general information) it is logical that Kiwix will immediately display both files that the user has just downloaded.

It is the user who must control the program, not the program must tell the user his preferences.

The program should not think that it is smarter than the user, it is the user who must tell the program what to do. A good program does not decide anything for the user, but only offers tools.

**Therefore, all default filters must be reset to display all data.**

> Guessing the language from the OS is a good approach because using an OS in a language you don't understand simply does not make sense.

Here I also completely disagree. When I install third-party software or even OS for my work, if possible, I always choose English. English is not my native language. But for finding a solution on the Internet, English is best suited.

You can display the system interface in the same language based on the OS of the system (this is logical), but in no case apply filters based based only on the system language.

**TL;DR - user can use multiple languages at the same time, so all filters should be reset to default rather than limit the user.**

**Comment by kelson42** [comment]
> > I disagree to display all languages per default
> 
> 
> 
> I don't agree with this. Let me explain my position: 
> 
> 
> 
> When a user downloads a ZIM file in their native language (because there is more complete local information) and a ZIM file in English (because there is more complete general information) it is logical that Kiwix will immediately display both files that the user has just downloaded.

If a user has achieved to download a content in English, then he will have to have the english language in the filter and it will stay that way without any contrary decision of the user. So both books will be visible.

> It is the user who must control the program, not the program must tell the user his preferences.
>
> The program should not think that it is smarter than the user, it is the user who must tell the program what to do. A good program does not decide anything for the user, but only offers tools.

It's never black/white like this, but here it's not about been smarter than the user, it's about default configuration before user has done any custom configuration. Keep in mind that we will keep all the configuration, so once the user will have made choices, it will remain.

> **Therefore, all default filters must be reset to display all data.**
> 
> 
> 
> > Guessing the language from the OS is a good approach because using an OS in a language you don't understand simply does not make sense.
> 
> 
> 
> Here I also completely disagree. When I install third-party software or even OS for my work, if possible, I always choose English. English is not my native language. But for finding a solution on the Internet, English is best suited.

**Comment by gmikhail** [comment]
> If a user has achieved to download a content in English, then he will have to have the english language in the filter and it will stay that way without any contrary decision of the user. So both books will be visible.

I would like to clarify that some users (like me, for example) download ZIM files directly from the official website using a torrent.

For example, if the Internet connection is unstable and constantly cuts out, then it is impossible to download a large file inside the app (there is a high possibility of data corruption during downloading). Or if it is convenient for the user to use torrent for the sake of automation and limiting the download speed. 

In this case, if you manually place the downloaded files in the correct folder, Kiwix will see them, but will not show them to the user.

That is why I am in favor of all filters being disabled by default.

---

Even on the example of this topic - the user has the system language set to English, but he downloaded the Spanish ZIM files. And when he opened the app, he does not understand why his downloaded ZIM files are not displayed. 

And he offered the same solution as me:

> I would say it would be less confusing to simply show all by default, considering the UI doesn't even tell that you're only viewing the system language by default.

So at least two people agree on this :)

---

> Keep in mind that we will keep all the configuration, so once the user will have made choices, it will remain.

Yes, saving preferences will partially solve the problem. But I'm still in favor of turning off all filters by default. 

Ideally, of course, Kiwix should itself determine that new ZIM files have appeared in the library folder and show the user that some of the ZIM files are hidden by filters. But I am a supporter of the KISS principle (Keep it Short and Simple) because often the simplest solution is the most effective. 

And in this case, the simplest and most effective solution is to simply reset all filters so that they display everything by default.

**Comment by mgautierfr** [comment]
I think we are in the middle of a "expert vs beginner design problem".

- Beginner probably what to search in their language as they probably don't know English. (BTW, we will not see them ask for change in github). If they what to search of languages, they can change the filter.
- Expert may understand english (or any other language) and more importantly, they can have a zim file by other way than kiwix-desktop (direct download, copy from usb flash drive, ...) and put it in the monitored directory.

What about two different default settings ?
- OS language for remote catalog (searching zim file to download)
- ALL for the local catalog (display what the user have locally)

**Comment by gmikhail** [comment]
> * OS language for remote catalog (searching zim file to download)

Only if the user is explicitly shown that the filter is currently active. For example, like on the official Kiwix website.

![2023-08-07 184130](https://github.com/kiwix/kiwix-desktop/assets/4838367/8fd25813-d39e-47e7-bc26-c35108e9001d)

In the current stable version of the Kiwix app, the filter is permanently collapsed, which is why many users do not know that it is active.

![kiwix](https://github.com/kiwix/kiwix-desktop/assets/4838367/d035e5de-4382-4d72-86f6-a446dc9a4558)

> * ALL for the local catalog (display what the user have locally)

Yes, that's exactly what I asked for.

**Comment by mgautierfr** [comment]
> Only if the user is explicitly shown that the filter is currently active

Agree. This should be fixed with issue https://github.com/kiwix/kiwix-desktop/issues/958
